src/Slices/ServiceInstanceDetails/UI/Components/AttributesComponents/AttributesEditor.tsx

```diff
@@ -48,8 +77,63 @@ export const AttributesEditor: React.FC<Props> = ({
     setSelectedSet(value);
   };
 
+  /**
+   * useCallback hook onEditorUpdate
+   *
+   * Try-catch to set the editorState with the updated data
+   *
+   * @param {string} value - the JSON editor stringified value
+   * @param {boolean} isValid - whether the editor contains known errors or not
+   */
+  const onEditorUpdate = useCallback(
+    (value: string, isValid: boolean) => {
+      try {
+        const parsed = JSON.parse(value);
+
+        setEditorState(parsed);
+        setIsEditorValid(isValid);
+      } catch (_error) {
+        setIsEditorValid(false);
+      }
+    },
+    [setEditorState, setIsEditorValid],
+  );
+
+  /**
+   * onConfirm async method sending the patch request to Expert edit the attributes
+   */
+  const onConfirm = async (): Promise<void> => {
+    const message = words("instanceDetails.API.message.update")(username);
+
+    const patchAttributes: ExpertPatchAttributes = {
+      comment: message,
+      attribute_set_name: selectedSet,
+      current_version: instance.version,
+      patch_id: uuidv4(),
+      edit: [
+        {
+          edit_id: `${instance.id}_version=${instance.version}`,
+          operation: "replace",
+          target: ".",
+          value: editorState,
+        },
+      ],
+    };
+
+    mutate(patchAttributes);
+  };
+
   useEffect(() => {
-    setEditorData(JSON.stringify(attributeSets[selectedSet], null, 2));
+    if (isError) {
+      setErrorMessage(error.message);
+    }
+    if (isSuccess) {
+      setIsModalOpen(false);
+    }
+  }, [isSuccess, isError, error]);
+
+  useEffect(() => {
```

Comment:
I think that could be moved to value calculated on render according to new dev docs of react where they encourage to revisit useEffects and not to overuse them - [link](https://react.dev/learn/you-might-not-need-an-effect)

I'm not familiar with the flow and it could be that changing that is not that simple, then it's fine for me

Reply:
If we keep it on initial calculation, the data will never be updated in the editor when the set is changed, or the version. For this specific case, there's no way around it.